tests/tests_vm_string.c

```diff
@@ -0,0 +1,194 @@
+    { NULL, NULL },
+};
+
+
+static MunitResult
+test_vm_strncmpl(const MunitParameter params[], void* user_data) {
+    const char* vm_strncmpl_params;
+    char* left_string;
+    char* right_string;
+    char** parsed_args;
+    int num_characters_to_count, parameters_length, last_comma_position;
+    int expected_result, predicted_result;
+    (void) user_data;
+
+    vm_strncmpl_params = munit_parameters_get(params, "vm_strncmpl_params");
+    // Let's first parse the params
+    parameters_length = strlen(vm_strncmpl_params);
+
+    parsed_args = arg_str_to_array_of_str(vm_strncmpl_params, '/');
+
+    expected_result = atoi(parsed_args[0]);
+    num_characters_to_count = atoi(parsed_args[1]);
+    left_string = parsed_args[2];
+    right_string = parsed_args[3];
+
+    // Now we have parsed the arguments, let's test our function
+    predicted_result = vm_strncmpl(
+        left_string, right_string, num_characters_to_count);
+
+    assert_int(expected_result, ==, predicted_result);
+
+    free(left_string);
+    free(right_string);
+    free(parsed_args[1]);
+    free(parsed_args[0]);
+    free(parsed_args);
+    return MUNIT_OK;
+}
+
+static char* vm_strcmpl_params[] = {
+    /* Expected result/ left string/ right string */
+    (char*) "1/ one/ one",
+    (char*) "1/ pato/ pato",
+    (char*) "0/ one/ nne",
+    (char*) "0/ democracia liberal/ patocracia liberal",
+    NULL
+};
+
+static MunitParameterEnum test_vm_strcmpl_params[] = {
+    { (char*) "vm_strcmpl_params", vm_strcmpl_params },
+    { NULL, NULL },
+};
+
+static MunitResult
+test_vm_strcmpl(const MunitParameter params[], void* user_data) {
+    const char* vm_strcmpl_params;
+    char* left_string;
+    char* right_string;
+    int parameters_length, last_comma_position;
+    int expected_result, predicted_result;
+    (void) user_data;
+    char** parsed_args;
+
+    vm_strcmpl_params = munit_parameters_get(params, "vm_strcmpl_params");
+    // Let's first parse the params
+    parameters_length = strlen(vm_strcmpl_params);
+
+    parsed_args = arg_str_to_array_of_str(vm_strcmpl_params, '/');
+
+    expected_result = atoi(parsed_args[0]);
+    left_string = parsed_args[1];
+    right_string = parsed_args[2];
+
+    // Now we have parsed the arguments, let's test our function
+    predicted_result = vm_strcmpl(
+        left_string, right_string);
+
+    assert_int(expected_result, ==, predicted_result);
+
+    free(parsed_args[0]);
+    free(parsed_args);
+    free(left_string);
+    free(right_string);
+    return MUNIT_OK;
+}
+
+static char* vm_strsplit_params[] = {
+    /*Expected Result/ Input String/ Delimiter/ 1-Started Index */
+    (char*) "Gosto de pão/ Gosto de pão, queijo e margarina./ ,/ 1",
+    (char*) "Gosto de pão/ Gosto de pão, presunto, queijo e margarina./ ,/ 1",
+    (char*) " presunto/ Gosto de pão, presunto, queijo e margarina./ ,/ 2",
+    (char*) " queijo e margarina./ Gosto de pão, presunto, queijo e margarina./ ,/ 3",
+
```

Comment:
Remove unnecessary newlines.

```suggestion
static char* vm_strsplit_params[] = {
    /*Expected Result/ Input String/ Delimiter/ 1-Started Index */
    (char*) "Gosto de pão/ Gosto de pão, queijo e margarina./ ,/ 1",
    (char*) "Gosto de pão/ Gosto de pão, presunto, queijo e margarina./ ,/ 1",
    (char*) " presunto/ Gosto de pão, presunto, queijo e margarina./ ,/ 2",
    (char*) " queijo e margarina./ Gosto de pão, presunto, queijo e margarina./ ,/ 3",
    NULL
};
```